CustomEventTest are unstable

Run on jenkins two builds a test fails in two different ways.

https://hq2.camunda.com/jenkins/zeebe/job/zeebe/job/972-logstreams-tests/2/console

```java
13:28:41 [INFO] Running io.zeebe.gossip.GossipJoinTest
13:28:41 [INFO] Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.722 s - in io.zeebe.gossip.GossipJoinTest
13:28:41 [INFO] Running io.zeebe.gossip.CustomEventTest
13:28:41 [ERROR] Tests run: 11, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.994 s <<< FAILURE! - in io.zeebe.gossip.CustomEventTest
13:28:41 [ERROR] shouldRemoveCustomEventListener(io.zeebe.gossip.CustomEventTest)  Time elapsed: 0.095 s  <<< FAILURE!
13:28:41 org.junit.ComparisonFailure: expected:<[2]> but was:<[0]>
13:28:41 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
13:28:41 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
13:28:41 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
13:28:41 	at io.zeebe.gossip.CustomEventTest.shouldRemoveCustomEventListener(CustomEventTest.java:285)
13:28:41 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
13:28:41 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
13:28:41 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
13:28:41 	at java.lang.reflect.Method.invoke(Method.java:498)
13:28:41 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
13:28:41 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
13:28:41 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
13:28:41 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
13:28:41 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
13:28:41 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
13:28:41 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
13:28:41 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
13:28:41 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
13:28:41 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
13:28:41 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
13:28:41 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
13:28:41 	at java.lang.Thread.run(Thread.java:748)
```
https://hq2.camunda.com/jenkins/zeebe/job/zeebe/job/972-logstreams-tests/1/console
```java

13:18:58 [INFO] Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.755 s - in io.zeebe.gossip.GossipJoinTest
13:18:58 [INFO] Running io.zeebe.gossip.CustomEventTest
13:18:58 [ERROR] Tests run: 11, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 11.095 s <<< FAILURE! - in io.zeebe.gossip.CustomEventTest
13:18:58 [ERROR] shouldRemoveCustomEventListener(io.zeebe.gossip.CustomEventTest)  Time elapsed: 10.014 s  <<< ERROR!
13:18:58 org.junit.runners.model.TestTimedOutException: test timed out after 10 seconds
13:18:58 	at sun.misc.Unsafe.park(Native Method)
13:18:58 	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
13:18:58 	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2163)
13:18:58 	at io.zeebe.util.sched.future.CompletableActorFuture.get(CompletableActorFuture.java:144)
13:18:58 	at io.zeebe.util.sched.future.CompletableActorFuture.get(CompletableActorFuture.java:118)
13:18:58 	at io.zeebe.util.sched.FutureUtil.join(FutureUtil.java:28)
13:18:58 	at io.zeebe.util.sched.future.CompletableActorFuture.join(CompletableActorFuture.java:224)
13:18:58 	at io.zeebe.transport.ServerTransport.close(ServerTransport.java:63)
13:18:58 	at io.zeebe.gossip.util.GossipRule.after(GossipRule.java:160)
13:18:58 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:50)
13:18:58 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
13:18:58 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
13:18:58 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
13:18:58 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
13:18:58 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
13:18:58 	at java.lang.Thread.run(Thread.java:748)
13:18:58 
```

--- a/gossip/src/test/java/io/zeebe/gossip/CustomEventTest.java
+++ b/gossip/src/test/java/io/zeebe/gossip/CustomEventTest.java
@@ -132,8 +132,8 @@ public class CustomEventTest {
     // when
     gossip1.getPushlisher().publishEvent(TYPE_1, PAYLOAD_1);
 
-    cluster.waitUntil(() -> gossip2.receivedCustomEvent(TYPE_1, gossip1));
-    cluster.waitUntil(() -> gossip3.receivedCustomEvent(TYPE_1, gossip1));
+    cluster.waitUntil(() -> customEventListener2.getInvocations().count() == 1);
+    cluster.waitUntil(() -> customEventListener3.getInvocations().count() == 1);
 
     // then
     assertThat(customEventListener2.getInvocations().count()).isEqualTo(1);
@@ -257,9 +257,8 @@ public class CustomEventTest {
     // when
     gossip1.getPushlisher().publishEvent(TYPE_1, PAYLOAD_1);
 
-    cluster.waitUntil(() -> gossip2.receivedCustomEvent(TYPE_1, gossip1));
-
     // then
+    cluster.waitUntil(() -> counter.get() == 3);
     assertThat(counter.get()).isEqualTo(3);
   }
 
@@ -279,9 +278,8 @@ public class CustomEventTest {
 
     gossip1.getPushlisher().publishEvent(TYPE_1, PAYLOAD_1);
 
-    cluster.waitUntil(() -> gossip2.receivedCustomEvent(TYPE_1, gossip1));
-
     // then
+    cluster.waitUntil(() -> counter.get() == 2);
     assertThat(counter.get()).isEqualTo(2);
   }
 

--- a/gossip/src/test/java/io/zeebe/gossip/GossipFailureDetectionTest.java
+++ b/gossip/src/test/java/io/zeebe/gossip/GossipFailureDetectionTest.java
@@ -17,24 +17,29 @@ package io.zeebe.gossip;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-import io.zeebe.UnstableTest;
 import io.zeebe.clustering.gossip.GossipEventType;
 import io.zeebe.clustering.gossip.MembershipEventType;
 import io.zeebe.gossip.protocol.MembershipEvent;
 import io.zeebe.gossip.util.GossipClusterRule;
 import io.zeebe.gossip.util.GossipRule;
-import org.junit.*;
-import org.junit.experimental.categories.Category;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
 import org.junit.rules.Timeout;
 
 public class GossipFailureDetectionTest {
+  private static final GossipConfiguration CONFIGURATION =
+      new GossipConfiguration().setSuspicionMultiplier(10);
+
   private GossipRule gossip1 = new GossipRule("localhost:8001");
   private GossipRule gossip2 = new GossipRule("localhost:8002");
   private GossipRule gossip3 = new GossipRule("localhost:8003");
 
-  @Rule public GossipClusterRule cluster = new GossipClusterRule(gossip1, gossip2, gossip3);
+  @Rule
+  public GossipClusterRule cluster =
+      new GossipClusterRule(CONFIGURATION, gossip1, gossip2, gossip3);
 
-  @Rule public Timeout timeout = Timeout.seconds(10);
+  @Rule public Timeout timeout = Timeout.seconds(20);
 
   @Before
   public void init() {
@@ -88,7 +93,6 @@ public class GossipFailureDetectionTest {
   }
 
   @Test
-  @Category(UnstableTest.class)
   public void shouldSpreadSuspectEvent() {
     // given
     cluster.interruptConnectionBetween(gossip3, gossip1);
@@ -97,28 +101,24 @@ public class GossipFailureDetectionTest {
     // when send SUSPECT event
     cluster.waitUntil(() -> gossip1.receivedMembershipEvent(MembershipEventType.SUSPECT, gossip3));
     cluster.waitUntil(() -> gossip2.receivedMembershipEvent(MembershipEventType.SUSPECT, gossip3));
-
-    assertThat(gossip1.hasMember(gossip3)).isTrue();
-    assertThat(gossip2.hasMember(gossip3)).isTrue();
   }
 
   @Test
-  @Category(UnstableTest.class)
   public void shouldSpreadConfirmEvent() {
     // given
     cluster.interruptConnectionBetween(gossip3, gossip1);
     cluster.interruptConnectionBetween(gossip3, gossip2);
 
-    // when send CONFIRM event after suspicion timeout
+    // when member is removed
+    cluster.waitUntil(() -> !gossip1.hasMember(gossip3));
+    cluster.waitUntil(() -> !gossip2.hasMember(gossip3));
+
+    // then confirm event was spread
     cluster.waitUntil(() -> gossip1.receivedMembershipEvent(MembershipEventType.CONFIRM, gossip3));
     cluster.waitUntil(() -> gossip2.receivedMembershipEvent(MembershipEventType.CONFIRM, gossip3));
-
-    assertThat(gossip1.hasMember(gossip3)).isFalse();
-    assertThat(gossip2.hasMember(gossip3)).isFalse();
   }
 
   @Test
-  @Category(UnstableTest.class)
   public void shouldCounterSuspectEventIfAlive() {
     // given
     cluster.interruptConnectionBetween(gossip3, gossip1);

--- a/gossip/src/test/java/io/zeebe/gossip/util/GossipClusterRule.java
+++ b/gossip/src/test/java/io/zeebe/gossip/util/GossipClusterRule.java
@@ -30,7 +30,7 @@ import org.junit.runner.Description;
 import org.junit.runners.model.Statement;
 
 public class GossipClusterRule implements TestRule {
-  private static final int MAX_CONDITION_RETRIES = 100;
+  private static final int MAX_CONDITION_RETRIES = 1000;
 
   private final ActorSchedulerRule actorScheduler;
   private final ControlledActorClock clock;
@@ -77,13 +77,17 @@ public class GossipClusterRule implements TestRule {
   }
 
   public void waitUntil(BooleanSupplier condition) {
+    waitUntil(condition, MAX_CONDITION_RETRIES);
+  }
+
+  public void waitUntil(BooleanSupplier condition, int retries) {
     int i = 0;
 
-    while (!condition.getAsBoolean() && i < MAX_CONDITION_RETRIES) {
+    while (!condition.getAsBoolean() && i < retries) {
       clock.addTime(configuration.getProbeIntervalDuration());
 
       try {
-        Thread.sleep(10L);
+        Thread.sleep(1L);
       } catch (InterruptedException e) {
       }
 

